### ISSUE TITLE
Can we set accepted_resource_roles by resource name(cpus, mem, ...)?

### ISSUE THREAD

**Issue by fengyehong** [mesosphere/marathon#2548]
I want to specify `accepted_resource_roles` in a more concrete&flexible way, for example:
`cpus:marathon mem:*,marathon`
so that marathon will take cpus for role `marathon`, while take mem/disk/ports of role `*` and role `marathon`

*Different accept policy for each type of resource.*

Why I want this: my cluster run cpu heavy tasks, and i run Marathon with mesos-role `marathon`. I need to constrain my marathon to take only cpus for role `marathon`, but i can use mem/disk/ports for role `marathon` and `*`

I cannot do this by now with `default_accepted_resource_roles` in cli, or `accepted_resource_roles` in app definition.

In source code I noticed marathon use set to store acceptable roles by now, if we want to define different role set for each resource, we need to change from Set to Map, with resource name as key and roles set as value.
But this will need to change CLI(input a Map for default_accepted_resource_roles) and app definition(change accepted_resource_roles to Map).

I'm not sure, Can we do this? what is the preferred way to do this?
Thanks

**Comment by kolloch** [comment]
Hi @fengyehong, if I understand it correctly, there are framework quotas in the Mesos pipeline which probably solve similar problems. @rukletsov, is that correct?

**Comment by fengyehong** [comment]
I config mesos slave with `--resources=` and my configuration looks like this:
`--resources=cpus(marathon1):8;cpus(marathon2):8;cpus(*):8`
So, first, some cpus are reserved explicitly, second, all other resources are shared and no quotas defined(so all `*`)

If I set marathon1 to accept resource for role `*` and `marathon1`, marathon1 can run well.
If I need to make marathon1 to use cpus for and only for role `marathon1`, I set default_accepted_resource_roles to `marathon1`, then marathon` can not take resource mem/ports/disk for role `*`, so no offer can be accepted :( I cannot do this by now.


Why we define only cpus quotas? As the cluster growing, slaves have different hardware configurations, it's a lot of work to define how much resources to reserve for each role(how much cpus, how much mem, how much disks, which port range),so we set only what we concerns and leave all the other resources shared, makes it easier.
